### DISCUSSION TITLE
Provide an option to add object definitions from TOM Explorer to an existing DAX script

### DISCUSSION THREAD

**Discussion by bernatagulloesbrina** [TabularEditor/TabularEditor3#1065] [Ideas]
It would be convenient for a number of things, including seeing the value expression and the format string expression of calculation items side by side for example

**Comment by otykier** [comment]
DAX Scripts will have support for calc groups and calc items in a future update (planned for release soon). This will allow you to view both the calc item expression and format string expression for multiple calc items in a single document, or open multiple DAX scripts to view them side by side. Do you think that will suffice?

See also #57

**Comment by bernatagulloesbrina** [comment]
I see!
Probably I need to get used to dax scripts. I seldom use them.

I think I need an easy way to add measures to existing dax scripts from the
model. is there one already or can only be done when you create a new dax
script?

**Comment by otykier** [comment]
Just start to type "MEA" (for `MEASURE`) and auto-complete will guide you:

![adding-measures-through-scripts](https://user-images.githubusercontent.com/8976200/135805590-ff065d93-630e-4e7b-a8c2-6388f932bb10.gif)

Once you hit F5, the measure will be added/updated in the model.

**Comment by bernatagulloesbrina** [comment]
Oh I see, but I meant to add the definitions of existing measures once the
script is created. The quickest way is to type the name of the measure and
right click on it and select add definition? I'd like to right click the
measure in the object model

**Comment by otykier** [comment]
> I'd like to right click the measure in the object model

This is a good feature request. Currently, clicking "Script DAX" on a measure would create a new script, which you could then copy into the existing one, but I agree that it'd be good to have an "Add measure to script" option if you already have a DAX script in front of you.

**Comment by bernatagulloesbrina** [comment]
Awesome, thanks!

**Comment by otykier** [comment]
@bernatagulloesbrina did you know that you can include definitions of measures by right-clicking on a measure reference in a DAX script (or DAX query)?

![image](https://user-images.githubusercontent.com/8976200/146454199-4e04b399-2d5a-40b1-815f-ed9d27a4a402.png)

As you can see there are two options - the first one will include the definition of the measure you right-clicked on, the other one will also include all dependencies of that measure.

**Comment by bernatagulloesbrina** [comment]
Awesome!
that's beautiful



El jue, 16 dic 2021 a las 22:51, Daniel Otykier ***@***.***>)
escribió:

> @bernatagulloesbrina <https://github.com/bernatagulloesbrina> did you
> know that you can include definitions of measures by right-clicking on a
> measure reference in a DAX script (or DAX query)?
>
> [image: image]
> <https://user-images.githubusercontent.com/8976200/146454199-4e04b399-2d5a-40b1-815f-ed9d27a4a402.png>
>
> As you can see there are two options - the first one will include the
> definition of the measure you right-clicked on, the other one will also
> include all dependencies of that measure.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TabularEditor/TabularEditor3/issues/238#issuecomment-996221254>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJUUXHLLUQMDRW4AFB34IQ3URJNPHANCNFSM5FGYG3CQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Bernat Agulló

Partner & Data Enthusiast

***@***.***

+34 637 93 42 76

Balmes, 177

08006 Barcelona
